### PR TITLE
An assortment of fixes

### LINKED ISSUE #693: Source distributed over fissionable zones
It is possible to create a distributed source using the flag `only_fissionable=True` over zones with no fissionable isotopes in them. In this case, OpenMC currently hangs on `Initializing source particles...` indefinitely.

If enough particles are rejected, it should probably raise an error and alert the user.

### LINKED ISSUE #701: Building with OpenMP without defining CC
I recently updated to the latest version of develop and ran into an issue building on INL's Falcon supercomputer. In particular, the issue only occurs when building with OpenMP. When I try to configure and build with the following commands:

```bash
> FC=mpif90 CC=mpicc HDF5_ROOT=/path/to/phdf5 cmake -Dopenmp=on ..
> make
```

I obtain the following error at link time:

```bash
[100%] Linking Fortran executable bin/openmc
CMakeFiles/openmc.dir/src/cmfd_input.F90.o: In function `__cmfd_input_MOD_read_cmfd_xml':
cmfd_input.F90:(.text+0x17da): undefined reference to `GOMP_parallel'
CMakeFiles/openmc.dir/src/eigenvalue.F90.o: In function `__eigenvalue_MOD_join_bank_from_threads':
eigenvalue.F90:(.text+0x269): undefined reference to `GOMP_parallel'
CMakeFiles/openmc.dir/src/global.F90.o: In function `__global_MOD_free_memory':
global.F90:(.text+0x118d): undefined reference to `GOMP_parallel'
CMakeFiles/openmc.dir/src/initialize.F90.o: In function `__initialize_MOD_allocate_banks':
initialize.F90:(.text+0x1e8f): undefined reference to `GOMP_parallel'
CMakeFiles/openmc.dir/src/input_xml.F90.o: In function `__input_xml_MOD_read_ce_cross_sections':
input_xml.F90:(.text+0x1eb5): undefined reference to `GOMP_parallel'
CMakeFiles/openmc.dir/src/mgxs_data.F90.o:mgxs_data.F90:(.text+0x5cd): more undefined references to `GOMP_parallel' follow
collect2: error: ld returned 1 exit status
make[2]: *** [bin/openmc] Error 1
make[1]: *** [CMakeFiles/openmc.dir/all] Error 2
make: *** [all] Error 2
```

After some trial and error I discovered that if I prepend `CC=mpicc` to the CMake command then the code compiles and links just fine. Has anyone else run into this? Is this an issue, or something that needs to be added to the documentation? I've not followed some of the major recent PRs (multipole, `openmc.data`) so I don't know what changed in the last two months to necessitate the use of `CC` when building OpenMC.

### PR DESCRIPTION
**Pull request by paulromano**
This pull request makes four minor fixes/improvements:

1. When a user specifies a bounding box for a `VolumeCalculation`, the API now tries to compute a bounding box for cells and if the user-specified box is smaller than the computed one, a warning is given.
2. The criterion for rejecting an external source has been changed. Now a fatal error is thrown only if 95%+ of sampled sites are rejected (also closes #693)
3. Update documentation to mention usage of CC environment variable. Closes #701.
4. Add SiO2 as a recognized S(a,b) material (for converting ACE -> HDF5).

### CODE REVIEW

**Comment by wbinventor** [comment]
This looks good to me.

**Comment by paulromano** [comment]
Two more things I just added:

- If you ask for a micro MGXS (either through `get_xs` or `get_pandas_dataframe`) and the domain is heterogeneous, atom densities determined from a volume calculation must be used. In this case, it's possible that the `domain.get_nuclides()` method has a larger set of nuclides than does `domain.get_nuclide_densities()` because the former searches the geometry recursively and the latter only has nuclides that were found in the volume calculation. If this happens, you end up getting a `ValueError` from `MGXS.get_nuclide_density`. I've changed this behavior so that it returns zero rather than raising an exception. I've also converted NaNs to zeros where a divide-by-zero occurs (if a nuclide density is zero).
- RELAX NG schema for settings.xml had `<volume_calc>` as one-or-more instead of zero-or-mode.

**Comment by paulromano** [comment]
@samuelshaner I just put a hotfix for a regression caused by #691. The `MGXS.get_pandas_dataframe()` method was modified such that the number of scores was assumed to be one -- however, `ScatterMatrixXS.get_pandas_dataframe()` calls this method, and if there are multiple Legendre moments, there is more than one score. This restores the earlier logic to account for the number of scores.

The test results that changed as a result of the hotfix are actually the same as they were before #691 was merged.

File: openmc/mgxs/mgxs.py [hunk 1]
```diff
@@ -1523,7 +1523,8 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
 
             df.rename(columns={'energyout low [MeV]': 'group out'},
                       inplace=True)
-            out_groups = np.tile(all_groups, int(df.shape[0] / all_groups.size))
+            out_groups = np.repeat(all_groups, self.xs_tally.num_scores)
+            out_groups = np.tile(out_groups, int(df.shape[0] / out_groups.size))
```
Comment (samuelshaner) [inline openmc/mgxs/mgxs.py]: Thanks for making this fix! The repeating/tiling can be confusing to wrap one's head around. Could you fix this in `mdgxs.py` on line 808 as well?

File: openmc/mgxs/mgxs.py [hunk 2]
```diff
@@ -548,26 +548,14 @@ def get_nuclide_density(self, nuclide):
         float
             The atomic number density (atom/b-cm) for the nuclide of interest
 
-        Raises
```
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: I'm curious if we should also return 0.0 from the `get_xs(...)` method for nuclides which are not found in an MGXS domain? I don't see a use case for this for myself right now, but perhaps the good folks at ANL-NE who are stressing this capability might.

File: src/source.F90 [hunk 3]
```diff
@@ -162,8 +156,21 @@ subroutine sample_external_source(site)
           end if
         end if
       end select
+
+      ! Check for rejection
+      if (.not. found) then
+        n_reject = n_reject + 1
+        if (n_reject >= EXTSRC_REJECT_THRESHOLD .and. &
+             real(n_sample, 8)/n_reject <= EXTSRC_REJECT_FRACTION) then
```
Comment (wbinventor) [inline src/source.F90]: Perhaps I'm confused, but won't `real(n_sample, 8) /n_reject` always be greater than unity? I *think* that `n_sample` will always be greater than or equal to `n_reject`, but I may be wrong. But if so, then I think the quotient and dividend should be interchanged to compute the rejection fraction.

**Comment by wbinventor** [comment]
Looks good to me. I'll merge once you take a look at my two comments.

File: src/source.F90 (hunk 3 above)
Comment (paulromano) [inline src/source.F90]: `n_sample` is really the number of *accepted* samples. I'll change the variable name so it's more obvious.

File: openmc/mgxs/mgxs.py (hunk 2 above)
Comment (paulromano) [inline openmc/mgxs/mgxs.py]: Looks like this would require changes in `Tally.get_values`. I'm going to defer for now until someone complains about it :)

**Comment by paulromano** [comment]
@samuelshaner I made the fix in mdgxs.py
@wbinventor Comments have been addressed

**Pull request merged**